lib/web_ui/lib/src/engine/safe_browser_api.dart

```diff
@@ -217,17 +217,26 @@ const bool _browserImageDecodingEnabled = bool.fromEnvironment(
 );
 
 /// Whether the current browser supports `ImageDecoder`.
-bool browserSupportsImageDecoder = _defaultBrowserSupportsImageDecoder;
+bool browserSupportsImageDecoder = _kDefaultBrowserSupportsImageDecoder;
 
 /// Sets the value of [browserSupportsImageDecoder] to its default value.
 void debugResetBrowserSupportsImageDecoder() {
-  browserSupportsImageDecoder = _defaultBrowserSupportsImageDecoder;
+  browserSupportsImageDecoder = _kDefaultBrowserSupportsImageDecoder;
 }
 
-bool _defaultBrowserSupportsImageDecoder =
+bool get _kDefaultBrowserSupportsImageDecoder =>
     _browserImageDecodingEnabled &&
     _imageDecoderConstructor != null &&
-    browserEngine == BrowserEngine.blink;
+    _kProtectFromImageDecoderIncompatibilities;
+
+// TODO(yjbanov): https://github.com/flutter/flutter/issues/122761
+// Frequently, when a browser launches an API that other browsers already
+// support, there are subtle incompatibilities that may cause apps to crash if,
+// we blindly adopt the new implementation. This variable prevents us from
+// picking up potentially incompatible implementations of ImagdeDecoder API.
+// Instead, when a new browser engine launches the API, we'll evaluate it and
+// enable it explicitly.
+bool get _kProtectFromImageDecoderIncompatibilities => browserEngine == BrowserEngine.blink;
```

Comment:
Complete nitpick, but I don't like the name of this variable. On its face, the name seems to imply the opposite of what it does. If something called `ProtectFromImageDecoderIncompatibilities` is true, I would think that means that you are protecting from incompatibilities, and if it's false, I would think you would not be protecting from incompatibilities. In general, either way, the name is a bit confusing as a name of a boolean. Might I suggest something like `_kImageDecoderIsStable` or `_kImageDecoderIsBlessed` or something like that?

Reply:
Agreed. I also removed the `k` prefix since these are getters and not consts anymore.